dot command does not support arguments in sh/dash

**Describe the bug**

dot command does not support arguments in sh/dash (see [SC2240](https://www.shellcheck.net/wiki/SC2240)). This shellcheck complain is incorrectly overwritten:

```
$ git grep SC2240
modules.d/99base/dracut-lib.sh:            # shellcheck disable=SC2240
modules.d/99shutdown/shutdown.sh:        # shellcheck disable=SC1090 disable=SC2240
```

`source_all` is only called by `source_hook`. Following `source_hook` specify parameters:

| module | call |
| --- | --- |
| `modules.d/35connman/cm-run.sh` | `source_hook initqueue/online "$ifname"` |
| `modules.d/35network-legacy/dhclient-script.sh` | `source_hook initqueue/online "$ifname"` |
| `modules.d/35network-legacy/ifup.sh` | `source_hook initqueue/online "$netif"` |
| `modules.d/35network-manager/nm-run.sh` | `source_hook initqueue/online "$ifname"` |
| `modules.d/45net-lib/netroot.sh` | `source_hook netroot "$netif"` |

**Distribution used**
Ubuntu

**Dracut version**
102

**Expected behavior**
All SC2240 should be fixed to support dash.

But still two places to fix:
```
$ git grep SC2240
modules.d/80base/dracut-lib.sh:        # shellcheck disable=SC2240
modules.d/86shutdown/shutdown.sh:        # shellcheck disable=SC1090 disable=SC2240
```

--- a/modules.d/80base/dracut-lib.sh
+++ b/modules.d/80base/dracut-lib.sh
@@ -394,9 +394,9 @@ source_hook() {
     _dir=$1
     shift
     for f in $(list_hooks "$_dir"); do
+        set -- "$@"
         # shellcheck disable=SC1090
-        # shellcheck disable=SC2240
-        . "$f" "$@"
+        . "$f"
     done
 }
 

--- a/modules.d/86shutdown/shutdown.sh
+++ b/modules.d/86shutdown/shutdown.sh
@@ -131,8 +131,11 @@ _check_shutdown() {
     local __f
     local __s=0
     for __f in $(list_hooks "shutdown"); do
-        # shellcheck disable=SC1090 disable=SC2240
-        if (final="$1" . "$__f" "$1"); then
+        if (
+            set -- "$1"
+            # shellcheck disable=SC1090
+            final="$1" . "$__f"
+        ); then
             rm -f -- "$__f"
         else
             __s=1

